src/clang.rs

```diff
@@ -645,19 +645,22 @@ impl Type {
     }
 
     /// If this type is a class template specialization, return its number of
-    /// template arguments. Otherwise, return -1.
-    pub fn num_template_args(&self) -> c_int {
-        unsafe {
-            clang_Type_getNumTemplateArguments(self.x)
+    /// template arguments. Otherwise, return None.
+    pub fn num_template_args(&self) -> Option<u32> {
+        let n = unsafe { clang_Type_getNumTemplateArguments(self.x) };
+        if n >= 0 {
+            Some(n as u32)
+        } else {
+            assert!(n==-1);
```

Comment:
nit: spacing is odd here (`n == -1`), but it'd be better if you do `assert_eq!(n, -1)`, or even better `debug_assert_eq!(n, -1)` so it gets compiled out in release builds.